implement hostnic

duanjiong opened the pull request:
Signed-off-by: Duan Jiong <djduanjiong@gmail.com>

duanjiong committed:
implement hostnic

Signed-off-by: Duan Jiong <djduanjiong@gmail.com>

zheng1 commented on config/dev/nfs_patch.yaml:
```diff
@@ -0,0 +1,22 @@
+apiVersion: apps/v1
```
Is it necessary to add this file?

duanjiong replied:
Not necessary, this is a dev, I use it for my own development

zheng1 commented on docs/usage.md:
```diff
@@ -0,0 +1,83 @@
+## 使用host
+
+* 给节点加上vxnet注解
+  `kubectl  annotate nodes i-xrwbww35  "hostnic.network.kubesphere.io/vxnet"="vxnet-cfn58ev"`
```
Is it possible to have a set of nodes or all nodes share a set of vxnet?

zheng1 replied:
All nodes share a set of vxnet without add any annotations.

duanjiong replied:
I've added this configuration before, but thought I'd remove it. A vxnet can create 254 NICs, each host can mount 60 NICs, and a vxnet can only be shared by four hosts, so it's easy to exceed the limit, so it's mandatory to use annotation selection here, so that users can also plan their use

duanjiong replied:
> Is it possible to have a set of nodes or all nodes share a set of vxnet?

Currently, only one NIC in vxnet is allowed to be cached, there is no support for caching multiple NICs, and there is no need to do so, as it will only increase the complexity of management and waste NIC resources.

duanjiong replied:
> All nodes share a set of vxnet without add any annotations.

I've added this configuration before, but thought I'd remove it. A vxnet can create 254 NICs, each host can mount 60 NICs, and a vxnet can only be shared by four hosts, so it's easy to exceed the limit, so it's mandatory to use annotation selection here, so that users can also plan their use

duanjiong committed:
Adapt dockershim to solve NIC leaks after node reboot pod rebuild

Add a configuration item to automatically annotate the node according to maxNic

Signed-off-by: Duan Jiong <djduanjiong@gmail.com>
